src/engine/EventManager.cxx

```diff
@@ -304,24 +341,35 @@ void EventManager::checkEvents(SDL_Event &event, Engine &engine)
           {
             m_nodesToPlace.push_back(mouseIsoCoords);
           }
-          bool placementAllowed = false;
+          placementAllowed = false;
+          std::vector<Point> nodesToAdd;
 
           // if we touch a bigger than 1x1 tile also add all nodes of the building to highlight.
           for (const auto &coords : m_nodesToHighlight)
           {
-            Layer layer = TileManager::instance().getTileLayer(tileToPlace);
+            const Layer layer = TileManager::instance().getTileLayer(tileToPlace);
             Point currentOriginPoint = engine.map->getNodeOrigCornerPoint(coords, layer);
+
+            if (currentOriginPoint == UNDEFINED_POINT)
+            { // This should never happen, but check we'll check it anyway just to be safe
+              continue;
+            }
+
             std::string currentTileID = engine.map->getTileID(currentOriginPoint, layer);
             for (auto &foundNode : engine.map->getObjectCoords(currentOriginPoint, currentTileID))
             {
               // only add the node if it's unique
-              if (std::find(m_nodesToHighlight.begin(), m_nodesToHighlight.end(), foundNode) == m_nodesToHighlight.end() && foundNode != UNDEFINED_POINT)
+              if (std::find(m_nodesToHighlight.begin(), m_nodesToHighlight.end(), foundNode) == m_nodesToHighlight.end() &&
+                  foundNode != UNDEFINED_POINT)
```

Comment:
the same as above.
will never happen becuase getObjectCoords will never return a container that contains UNDEFINED_POINT.

Reply:
You're right. This happened before i fixed the for loop, because i was modifying the container that was used in the for loop.